mesonpy/__init__.py

```diff
@@ -400,7 +400,7 @@ def _warn_unsure_platlib(self, origin: pathlib.Path, destination: pathlib.Path)
         """
         # {moduledir_shared} is currently handled in heuristics due to a Meson bug,
         # but we know that files that go there are supposed to go to platlib.
-        if self._is_native(origin):
+        if origin.is_fifo() and self._is_native(origin):
```

Comment:
If anything, this should be `origin.is_file()`

Reply:
Yes, sorry, wrong autocomplete.